src/tapeimage.cpp

```diff
@@ -525,14 +522,16 @@ std::int64_t tapeimage::readinto(void* dst, std::int64_t len) noexcept (false) {
 
 // TODO: status instead of boolean?
 int tapeimage::eof() const noexcept (true) {
-    assert(not this->index.empty());
     // TODO: consider when this says record, but physical file is EOF
     // TODO: end-of-file is an _empty_ record, i.e. two consecutive tape marks
     return this->current->type == tapeimage::file;
 }
 
 void tapeimage::read_header_from_disk() noexcept (false) {
-    assert(this->index.empty() or this->current == this->index.last());
+    try {
+        assert(this->index.last()->next == this->fp->tell());
+    } catch (const lfp::error&) {
```

Comment:
Maybe comment which error you are expecting here to ignore?
It's not very obvious.